src/Controller/EthereumController.php

```diff
@@ -21,15 +23,20 @@ class EthereumController extends ControllerBase {
    *
    * @var \Ethereum\Ethereum
    */
-  protected $client;
+  public $client;
 
   private $debug = TRUE;
 
   /**
    * Constructs a new EthereumController.
+   *
+   * @param \Ethereum\Ethereum $web3
    */
-  public function __construct(Ethereum $ethereum_client) {
-    $this->client = $ethereum_client;
+  public function __construct(Ethereum $web3 = NULL) {
```

Comment:
This change should not be needed since we're are injecting the ethereum client from the container.

I assume that some other code is not initializing this class properly..

Reply:
Hopefully fixed. This topic is still confusing for me.